### DISCUSSION TITLE
aws ec2 nvme disks

### DISCUSSION THREAD

**Discussion by mzupan** [bottlerocket-os/bottlerocket#3545] [Q&A]
Anyone have a good way for bottlerocket to be able to find and format a NVME local SSD to use as a volume?

**Comment by stmcginnis** [comment]
Hey @mzupan - good question. The recommended way for doing custom storage configuration would be to use a [bootstrap container](https://github.com/bottlerocket-os/bottlerocket#bootstrap-containers-settings). That runs early on and would allow you to configure things before most services have started up.

There's a good example of doing something similar here: https://github.com/bottlerocket-os/bottlerocket/discussions/1991#discussioncomment-3265188

You would want to review the example script given to adjust for your given use case, but hopefully that gives a good start and example for what would need to be done.
